test/global.test.ts

```diff
@@ -26,8 +39,39 @@ suiteSetup(async function (this: Mocha.Context): Promise<void> {
         await extension.activate();
     }
 
+    await fse.ensureDir(testFolderPath);
+    await fse.emptyDir(testFolderPath);
+    testWorkspacePath = await initTestWorkspacePath();
+
     ext.outputChannel = new TestOutputChannel();
     ext.ui = testUserInput;
 
     longRunningTestsEnabled = !/^(false|0)?$/i.test(process.env.ENABLE_LONG_RUNNING_TESTS || '');
+
+    // set the user name and email since this seems to be required for everything
+    await cpUtils.executeCommand(undefined, undefined, 'git', 'config', '--global', 'user.name', 'Automated Tester');
+    await cpUtils.executeCommand(undefined, undefined, 'git', 'config', '--global', 'user.email', 'automated@testing.com');
 });
+
+suiteTeardown(async function (this: Mocha.Context): Promise<void> {
+    await fse.emptyDir(testFolderPath);
+    await fse.remove(testFolderPath);
+});
+
+export async function cleanTestWorkspace(): Promise<void> {
```

Comment:
If you're not gonna use `testWorkspacePath`, I don't think you need to add that code. If you are going to use it, you need to call `cleanTestWorkspace` in `suiteTeardown`